kcbq-connector/src/test/java/com/wepay/kafka/connect/bigquery/BigQuerySinkTaskTest.java

```diff
@@ -123,13 +125,91 @@ public void testSimplePutWhenSchemaRetrieverIsNotNull() {
+    SchemaManager schemaManager = mock(SchemaManager.class);
+
+    BigQuerySinkTask testTask = new BigQuerySinkTask(bigQuery, schemaRetriever, storage, schemaManager);
+    testTask.initialize(sinkTaskContext);
+    testTask.start(properties);
+
+    testTask.put(Collections.singletonList(spoofSinkRecord(nonExistingTableTopic)));
+    testTask.flush(Collections.emptyMap());
+
+    verify(schemaManager, never()).createTable(existingTable, existingTableTopic);
+    verify(schemaManager).createTable(nonExistingTable, nonExistingTableTopic);
+  }
+
+  @Test
+  public void testNonAutoCreateTables() {
+    final String dataset = "scratch";
+    final String existingTableTopic = "topic-with-existing-table";
+    final String nonExistingTableTopic = "topic-without-existing-table";
+    final TableId existingTable = TableId.of(dataset, "topic_with_existing_table");
+    final TableId nonExistingTable = TableId.of(dataset, "topic_without_existing_table");
+
+    Map<String, String> properties = propertiesFactory.getProperties();
+    properties.put(BigQuerySinkConfig.SCHEMA_RETRIEVER_CONFIG, BigQuerySinkConnectorTest.MockSchemaRetriever.class.getName());
+    properties.put(BigQuerySinkConfig.SANITIZE_TOPICS_CONFIG, "true");
+    properties.put(BigQuerySinkConfig.DATASETS_CONFIG, String.format(".*=%s", dataset));
+    properties.put(BigQuerySinkConfig.TOPICS_CONFIG, existingTableTopic);
+
+    BigQuery bigQuery = mock(BigQuery.class);
+    Table fakeTable = mock(Table.class);
+    when(bigQuery.getTable(existingTable)).thenReturn(fakeTable);
+    when(bigQuery.getTable(nonExistingTable)).thenReturn(null);
+    InsertAllResponse insertAllResponse = mock(InsertAllResponse.class);
+    when(bigQuery.insertAll(anyObject())).thenReturn(insertAllResponse);
+    when(insertAllResponse.hasErrors()).thenReturn(false);
+
+    Storage storage = mock(Storage.class);
+    SchemaRetriever schemaRetriever = mock(SchemaRetriever.class);
+    SinkTaskContext sinkTaskContext = mock(SinkTaskContext.class);
+    SchemaManager schemaManager = mock(SchemaManager.class);
+
+    BigQuerySinkTask testTask = new BigQuerySinkTask(bigQuery, schemaRetriever, storage, schemaManager);
+    testTask.initialize(sinkTaskContext);
+    testTask.start(properties);
+
+    testTask.put(Collections.singletonList(spoofSinkRecord(nonExistingTableTopic)));
+    testTask.flush(Collections.emptyMap());
+
+    verify(schemaManager, never()).createTable(existingTable, existingTableTopic);
+    verify(schemaManager, never()).createTable(nonExistingTable, existingTableTopic);
+  }
+
+
   @Test
   public void testEmptyPut() {
     Map<String, String> properties = propertiesFactory.getProperties();
     BigQuery bigQuery = mock(BigQuery.class);
     Storage storage = mock(Storage.class);
 
-    BigQuerySinkTask testTask = new BigQuerySinkTask(bigQuery, null, storage);
+    BigQuerySinkTask testTask = new BigQuerySinkTask(bigQuery, null, storage, null);
```

Comment:
Suggest making a second constructor for `BigQuerySinkTask` that doesn't require test classes to be specified (just defaults them to null.

Reply:
There should already be one (line 97 ~ line 105):
```
 /**
   * Create a new BigquerySinkTask.
   */
  public BigQuerySinkTask() {
    testBigQuery = null;
    schemaRetriever = null;
    testGcs = null;
    testSchemaManager = null;
  }```